File: examples/java-ddd/bootstrap/src/test/java/org/myddd/java/bootstrap/controller/UserControllerTest.java
package org.myddd.java.bootstrap.controller;

import com.google.gson.JsonObject;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.myddd.java.api.dto.UserDTO;
import org.myddd.java.bootstrap.AbstractControllerTest;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;

public class UserControllerTest extends AbstractControllerTest {

    @Test
    void createUser(){
        ResponseEntity<UserDTO> errorResponse = restTemplate.exchange(baseUrl()+"/v1/users", HttpMethod.POST,createHttpEntityFromString(randomUserJSON()),UserDTO.class);
        Assertions.assertTrue(errorResponse.getStatusCode().is2xxSuccessful());
    }


    private String randomUserJSON(){
        var jsonObject = new JsonObject();
        jsonObject.addProperty("userId",randomString());
        jsonObject.addProperty("name",randomString());
        jsonObject.addProperty("phone",randomString());
        jsonObject.addProperty("email",randomString());
        return jsonObject.toString();
    }
}

File: examples/kotlin-ddd/bootstrap/src/test/kotlin/org/myddd/kotlin/controller/UserControllerTest.kt
package org.myddd.kotlin.controller

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.myddd.kotlin.AbstractControllerTest
import org.myddd.kotlin.api.dto.UserDTO
import org.myddd.utils.Page

class UserControllerTest:AbstractControllerTest() {

    @Test
    fun createUser(){
        val responseEntity = restTemplate.postForEntity(baseUrl() + "/v1/users", createHttpEntityFromString(randomCreateUserJSON()),UserDTO::class.java)
        Assertions.assertTrue(responseEntity.statusCode.is2xxSuccessful)
    }

    @Test
    fun searchUser(){
        var responseEntity = restTemplate.getForEntity(baseUrl() + "/v1/users?search=${randomString()}", Page::class.java)
        Assertions.assertTrue(responseEntity.body.data.isEmpty())

        responseEntity = restTemplate.getForEntity(baseUrl() + "/v1/users", Page::class.java)
        Assertions.assertTrue(responseEntity.body.data.isEmpty())
    }

    private fun randomCreateUserJSON():String {
        return """
            {
              "name":"${randomString()}",
              "userId":"${randomString()}",
              "phone":"18620501006",
              "email":"${randomString()}"
            }
        """.trimIndent()
    }

}